Qt/mainwindow.h

```diff
@@ -110,23 +108,47 @@ private slots:
 	}
 	void linearAct() { g_Config.iTexFiltering = (g_Config.iTexFiltering != 0) ? 0 : 3; }
 
-	void postprocessingShaderGroup_triggered(QAction *action) { g_Config.sPostShaderName = action->data().toInt(); }
-	void renderingResolutionGroup_triggered(QAction *action) { g_Config.iInternalResolution = action->data().toInt(); }
+	void renderingResolutionGroup_triggered(QAction *action) {
+		g_Config.iInternalResolution = action->data().toInt();
+		NativeMessageReceived("gpu_resized", "");
+		if (g_Config.iTexScalingLevel == TEXSCALING_AUTO) {
+			NativeMessageReceived("gpu_clearCache", "");
+	}
+	}
 	void windowGroup_triggered(QAction *action) { SetWindowScale(action->data().toInt()); }
 
 	void displayLayoutGroup_triggered(QAction *action) {
 		g_Config.iSmallDisplayZoomType = action->data().toInt();
 		NativeMessageReceived("gpu_resized", "");
 	}
-	void renderingModeGroup_triggered(QAction *action) { g_Config.iRenderingMode = action->data().toInt(); }
-	void autoframeskipAct() { g_Config.bAutoFrameSkip = !g_Config.bAutoFrameSkip; }
+	void renderingModeGroup_triggered(QAction *action) {
+		g_Config.iRenderingMode = action->data().toInt();
+		g_Config.bAutoFrameSkip = false;
+		NativeMessageReceived("gpu_resized", "");
+	}
+	void autoframeskipAct() {
+		g_Config.bAutoFrameSkip = !g_Config.bAutoFrameSkip;
+		if (g_Config.iRenderingMode == FB_NON_BUFFERED_MODE) {
+			g_Config.iRenderingMode = FB_BUFFERED_MODE;
+			NativeMessageReceived("gpu_resized", "");
+	}
```

Comment:
nit: Indentation fuax pas.

-[Unknown]

Reply:
nit: faux pas doesn't mean mistake or slip-up, it means something embarrassing or socially unacceptable - though maybe indentation mistakes are ;)

Reply:
Yep, I say this just like I say "nit: Tabs and spaces living together in despicable harmony."  I'm not being serious or accusing anyone of some actual unacceptable act.

-[Unknown]